common/djangoapps/embargo/tests/test_api.py

```diff
@@ -0,0 +1,147 @@
+"""
+Tests for EmbargoMiddleware
+"""
+
+import mock
+import pygeoip
+import unittest
+
+from django.core.urlresolvers import reverse
+from django.conf import settings
+from django.test.utils import override_settings
+import ddt
+
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.factories import CourseFactory
+from xmodule.modulestore.tests.django_utils import (
+    ModuleStoreTestCase, mixed_store_config
+)
+
+# Explicitly import the cache from ConfigurationModel so we can reset it after each test
+from config_models.models import cache
+from embargo.models import EmbargoedCourse, EmbargoedState, IPFilter, RestrictedCourse, Country, CountryAccessRule, \
+    WHITE_LIST, BLACK_LIST
+from django_countries import countries
+
+
+# Since we don't need any XML course fixtures, use a modulestore configuration
+# that disables the XML modulestore.
+MODULESTORE_CONFIG = mixed_store_config(settings.COMMON_TEST_DATA_ROOT, {}, include_xml=False)
+
+
+@ddt.ddt
+@override_settings(MODULESTORE=MODULESTORE_CONFIG)
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class EmbargoApiTests(ModuleStoreTestCase):
+    """
+    Tests of EmbargoApi
+    """
+
+    def setUp(self):
+        self.user = UserFactory(username='fred', password='secret')
+        self.client.login(username='fred', password='secret')
+        self.embargo_course1 = CourseFactory.create()
+        self.embargo_course1.save()
+        self.embargo_course2 = CourseFactory.create()
+        self.embargo_course2.save()
+        self.regular_course = CourseFactory.create(org="Regular")
+        self.regular_course.save()
+        self.embargoed_course_whitelisted = '/courses/' + self.embargo_course1.id.to_deprecated_string() + '/info'
+        self.embargoed_course_blacklisted = '/courses/' + self.embargo_course2.id.to_deprecated_string() + '/info'
+        self.regular_page = '/courses/' + self.regular_course.id.to_deprecated_string() + '/info'
+
+        restricted_course_1 = RestrictedCourse(course_key=self.embargo_course1.id)
+        restricted_course_1.save()
```

Comment:
You should be able to create and save the model in one line instead of two:
```
restricted_course_1 = RestrictedCourse.objects.create(course_key=self.embargo_course1.id)
```